src/middlewared/middlewared/scripts/vendor_service.py

```diff
@@ -0,0 +1,62 @@
+import hashlib
+import subprocess
+import sys
+
+from truenas_api_client import Client
+
+
+def get_hostid():
+    try:
+        with open("/etc/hostid", "rb") as f:
+            return hashlib.file_digest(f, "sha256").hexdigest()
+    except:
+        pass
+
+
+def main():
+    with Client() as client:
+        vendor_name = client.call("system.vendor.name")
+        if vendor_name == "HexOS":
+            url = "wss://api.hexos.com"
+            systemd_opts = (
```

Comment:
Should there be a `Restart` parameter as well so that it is restarted if it fails for some reason?